src/namespace/controller.ts

```diff
@@ -0,0 +1,34 @@
+import { NodeDef, NodeAPI, Node, NodeMessageInFlow } from "node-red";
+import { ClusterConfigProperties } from "../cluster-config/controller";
+
+export interface NamespaceProperties extends NodeDef {
+    cluster: ClusterConfigProperties;
+    namespacename: string;
+}
+
+class NamespaceController {
+    node: Node;
+    cluster: ClusterConfigProperties;
+    namespacename: string;
+
+    constructor(node: Node, RED: NodeAPI, config: NamespaceProperties) {
+        node.on('input', this.onInput.bind(this));
+        this.node = node;
+        this.namespacename = config.namespacename;
+        var confignode = RED.nodes.getNode("cluter-config");
```

Comment:
wrong place but not sure how to get config-node yet....

Reply:
We have to do a bit more research here. the config node stuff is more about sharing configured properties rather than runtime state (in this case kube client) that derives from it. I do see the kube client ID in the debugger appearing in the node runtime but that's just a string 😅 

Let us investigate in other prior art, i.e. https://github.com/zachowj/node-red-contrib-home-assistant-websocket how it is done there.

Reply:
wrong statement: config nodes indeed can carry state, i.e. clients. Example in the core MQTT integration in NodeRed https://github.com/node-red/node-red/blob/master/packages/node_modules/%40node-red/nodes/core/network/10-mqtt.js#L1272 where a "MQTT out" node uses the "MQTT broker" node. The former is a node that emits events from a topic, the latter is a central config node.